### ISSUE TITLE
Gif support for ButtonBase, Image, Arrangement & ImageSprite

### ISSUE THREAD

**Issue by ColinTree** [mit-cml/appinventor-sources#1048]
It's quite necessary to support gif in appinventor. Many people asked me about animation supporting. And the currently the only solution is to switch pictures by a clock.

**Comment by ewpatton** [comment]
A quick search for GIF support in Android leads me to believe that this might require a min SDK of 14. The MIT team isn't ready to increase the API version that high just yet (production is at 4 and test is at 7). There might be a way to handle this gracefully without bumping the min SDK.

**Comment by ewpatton** [comment]
[for future reference](https://github.com/koral--/android-gif-drawable)

**Comment by ColinTree** [comment]
What about switching BitmapDrawable? Not graceful enough 😂

**Comment by ewpatton** [comment]
@ColinTree Switching to what, specifically? BitmapDrawable doesn't handle animations--it takes a single Bitmap and so cannot be used for rendering animations.

**Comment by ColinTree** [comment]
I was talking about switching between BDs which represent frames in gif file.

**Comment by ewpatton** [comment]
I agree that it would be backwards compatible to do it that way. You would still need code to handle decoding each frame, which would need to be written and maintained. It is better to use an already tested library rather than writing a solution from scratch. A simpler alternative would be to only support GIFs if the min SDK is >= 14, e.g. via the appropriate utility class.

**Comment by ColinTree** [comment]
I remember that you told us that there is a new companion and a upgrade of the sdk level before. what happen to this? it was so unnecessary to keep apps running on such an old system. Inmy opinion, with such a old fashion set of UI (by android 2), it is hard to increase awareness of appinventor.

**Comment by ewpatton** [comment]
The components release with the min SDK increase is scheduled for December 27.

Most of our users are students in schools, and the schools cannot necessarily afford to upgrade hardware every 2-3 years. We still have a number of users on Android 2 and 3, so it is necessary for us to move cautiously and maintain backwards compatibility for longer than is typical of most applications.

**Comment by ColinTree** [comment]
Okay got that

**Comment by ewpatton** [comment]
@mit-cml/cornell-cs-5152-students-fall-2019 For those of you interested in some mobile development experience this might be a fun one to work on (and I know a lot of high school students who would be grateful for it).

**Comment by kurtfchua** [comment]
@ewpatton Hi Evan can I take this?

**Comment by ewpatton** [comment]
@kurtfchua Here's a library you can use: https://github.com/bumptech/glide. I would recommend looking at v3 for maximum compatibility with our code base.

**Comment by jsuyash1514** [comment]
@ewpatton I would like to work on this issue.
